sw/device/tests/entropy_src_fw_observe_many_contiguous.c

```diff
@@ -0,0 +1,170 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_entropy_src.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/entropy_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kEntropySrcHealthTestWindowSize = 0x60,
+  /**
+   * Observe FIFO threshold: half of the FIFO size.
+   */
+  kEntropySrcFifoThreshold = 32,
+  /**
+   * The number of contiguous samples we want to capture.
+   */
+  kContiguousSamplesCount = 1024,
+  /*
+   * Timeout to read kContiguousSamplesCount.
+   */
+  kTimeoutUsec = 1000 * 1000,
+  /**
+   * Number of time to repeat each test, letting the observe FIFO
+   * overflow in-between tests.
+   */
+  kRepeatCount = 4,
+  /**
+   * Size of buffer is words to hold all the samples, assuming
+   * 4-bit samples at the most.
+   */
+  kFifoBufferSizeWords = kContiguousSamplesCount * 4 / sizeof(uint32_t),
+};
+
+uint32_t sample_buffer[kFifoBufferSizeWords];
+
+/**
+ * Test the firmware override observe path.
```

Comment:
Note that this test now disables the entropy_src and re-enables it again without disabling also CSRNG and the EDNs futher downstream. As mandated in the programmer's guide for the various modules, they need to be disabled / re-enabled in sync to avoid lockups (see e.g. [here for CSRNG](https://opentitan.org/book/hw/ip/csrng/doc/programmers_guide.html#module-enable-and-disable)).

This might be okay for this particular test as we don't check / plan to check whether the Firmware Override: Observe mode works while other peripherals are busy. So for this test it's more kind of a heads up if things get stuck it might be that some peripherals software is accidentally waiting for hangs due to not getting entropy from the EDNs.

For the other test where we want to test that peripherals can work with the entropy_src in Firmware Override: Extract and Insert mode, this will matter I guess :-)